lib/rdoc/markdown.kpeg

```diff
@@ -496,6 +496,46 @@
       "<s>#{text}</s>"
     end
   end
+
+  ##
+  # Parses inline markdown in table cells
+
+  def parse_table_cells(table)
+    # Parse header cells
+    table.header = table.header.map { |cell| parse_cell_inline(cell) }
+
+    # Parse body cells
+    table.body = table.body.map do |row|
+      row.map { |cell| parse_cell_inline(cell) }
+    end
+
+    table
+  end
+
+  ##
+  # Parses inline markdown in a single table cell
+
+  def parse_cell_inline(text)
+    return text if text.nil? || text.empty?
+
+    # Create a new parser instance for the cell
+    cell_parser = RDoc::Markdown.new(@extensions, @debug)
+
+    # Parse the cell content
+    doc = cell_parser.parse(text)
+
+    # Extract the parsed content
+    if doc && doc.parts && !doc.parts.empty?
+      para = doc.parts.first
+      if para.respond_to?(:parts)
```

Comment:
I think `para.is_a?(RDoc::Markup::Paragraph)` is better.

Some other class has parts, and it might not be an array of String.
```ruby
RDoc::Markdown.new.parse("> - a").parts.first.parts.join #=> "#<RDoc::Markup::List:0x000000013abea5d8>"
RDoc::Markdown.new.parse("```ruby\na\n```").parts.first.parts.join #=> "a\n"
```

Reply:
Good catch. Updated 👍